test/e2e-common.sh

```diff
@@ -327,6 +327,8 @@ function test_setup() {
 
   # Capture all logs.
   kail > ${ARTIFACTS}/k8s.log.txt &
+  # Clean up kail so it doesn't interfere with job shutting down
+  trap "[[ -n \"$(jobs | grep -E 'Running +kail')\" ]] && pkill kail || true" EXIT
```

Comment:
Just capture the PID ($!) and use it. Something like

KAIL_PID=$!
trap "kill $KAIL_PID" EXIT

Reply:
cool, updated, thanks!